app/controllers/events_controller.rb

```diff
@@ -254,6 +256,13 @@ def event_params
 
     end
 
+    def add_event_query_conditions(query)
```

Comment:
I'm not too happy with the naming here, but then again, I don't have any better ideas, either.